src/compiler/scala/reflect/quasiquotes/Holes.scala

```diff
@@ -132,7 +132,7 @@ trait Holes { self: Quasiquotes =>
     private def mapF(tree: Tree, f: Tree => Tree): Tree =
       if (f(Ident(TermName("x"))) equalsStructure Ident(TermName("x"))) tree
       else {
-        val x: TermName = c.freshName()
+        val x: TermName = TermName(c.freshName())
```

Comment:
you could remove the type annotation, it was only there to drive implicit search